### PR TITLE
Fix ungraded surveys not being shown in todo list

### PR DESCRIPTION
**Pull request by szabinst**
Include ungraded quizzes and surveys in ToDos screen.

refs: MBL-17750
affects: Student
release note: Ungraded quizzes and surveys now show up in ToDo.
test plan: See ticket

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [x] Follow-up e2e test ticket created
- [x] Tested on phone
- [x] Tested on tablet

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
#### Release Note: 
Ungraded quizzes and surveys now show up in ToDo.

#### Affected Apps: Student

[MBL-17750](https://instructure.atlassian.net/browse/MBL-17750)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.61% | 91.63% | -0.03%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against dad8ee1fa55d913682599a20e30375071042e9f3
</p>

File: Core/Core/Todos/Todo.swift [hunk 1]
```diff
@@ -59,12 +120,23 @@ public final class Todo: NSManagedObject, WriteableModel {
...
+            assignment.update(fromApiModel: apiAssignment, in: context, updateSubmission: false, updateScoreStatistics: false)
+            model.assignment = assignment
+            model.name = assignment.name
+            model.dueAtSortNilsAtBottom = assignment.dueAt ?? .distantFuture
+        } else if let apiQuiz = item.quiz {
+            let quiz: Quiz = Quiz.save(apiQuiz, in: context)
+            quiz.courseID = item.course_id?.value ?? "unknown"
+            model.quiz = quiz
+            model.name = quiz.title
+            model.dueAtSortNilsAtBottom = quiz.dueAt ?? .distantFuture
```
Comment (vargaat) [inline Core/Core/Todos/Todo.swift:137]: Just to double check: these are full quiz and assignment objects and not missing any fields that could be reset from the DB while saving these entities?

**Review by vargaat — APPROVED** [review]
QA+1

File: Core/Core/Todos/Todo.swift (hunk 1 above)
Comment (szabinst) [reply Core/Core/Todos/Todo.swift:137]: I double checked the quiz api they are full objects. The assignment fetching didn't change with this PR but I noticed the `assignment_groups` api doesn't return assignments with due date. Not sure if this is intentional API design but it's the same in the app store version.

**Review by ndrsszsz — APPROVED** [review]
QA +1
Tested on iPad, iOS 16.7.1

**Pull request merged**